### PR TITLE
Implemented remediation service to only provide next actions

### LINKED ISSUE #3682: Clean-up remediation-service to not control the remediation sequence
**Current status:** In its current implementation, the remediation-service controls the execution of the remediation sequence by sending out the `action.triggered` event and waiting for `action.finished` and `evaluation.finished`. Besides, waiting for 10 minutes is hard-coded into the services. 

Should we call the task: `get-action` ? Since we have the same discussion here: https://github.com/keptn/spec/issues/57 --> `get-action`

**Clean-up remediation-service:** The remediation-service does not control the sequence, but:
* it subscribes to a `get-action.triggered` event,
* it analysis the event payload and determines the next action to execute by looking into the `remediation.yaml`.
* If the `remediation.yaml` defines multiple actions for one problem and the one was already be executed, the remediation-service must be smart enough to take the next one in the list of actions. 

**Clean-up of spec:**
- By implementing this approach, the reserved Keptn Task `remediation` is not needed anymore: https://github.com/keptn/spec/blob/master/shipyard.md#remediation

As a result of this clean-up task, the shipyard-controller takes over the responsibility for sending `action.triggered`, and `evaluation.triggered` - since this is modeled in the shipyard. 

**Design decisions:**
* The [sh.keptn.event.problem.open](https://github.com/keptn/spec/blob/master/cloudevents.md#problem) event does not work as trigger for the `remediation` sequence --> it will be the `sh.keptn.event.remediation.triggered` that kicks off the sequence. 
* Can the issue https://github.com/keptn/spec/issues/59 be resolved then?

**Definition of Done**: 
- [ ] Adaptation of integration tests
- [ ] The remediation sequence execution is explicitly modeled by the shipyard, e.g.:
```
- name: "production" 
  sequences: 
  - name: remediation
      tasks:
      - name: get-action
      - name: action
      - name: evaluation
        triggeredAfter: "10m"
```

### PR DESCRIPTION
**Pull request by warber**
closes #3682 

Signed-off-by: warber <bernd.warmuth@dynatrace.com>

### CODE REVIEW

**Review by bacherfl — DISMISSED** [review]

**Comment by warber** [comment]
PR to be merged once the related issues are completed:
https://github.com/keptn/keptn/issues/3698
https://github.com/keptn/keptn/issues/3991
https://github.com/keptn/keptn/issues/3683
https://github.com/keptn/examples/issues/165

File: remediation-service/Dockerfile [hunk 1]
```diff
@@ -19,44 +20,30 @@ COPY go.mod go.sum ./
...
+
+# Build the command inside the container.
 # (You may fetch or manage dependencies here, either manually or with a tool like "godep".)
-RUN GOOS=linux go build -ldflags '-linkmode=external' $BUILDFLAGS -v -o remediation-service
+RUN GOOS=linux go build -ldflags '-linkmode=external' -gcflags="${SKAFFOLD_GO_GCFLAGS}" -v -o remediation-service
 
 # Use a Docker multi-stage build to create a lean production image.
 # https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds
-FROM alpine:3.13
+FROM alpine:3.11
```
Comment (christian-kreuzberger-dtx) [inline remediation-service/Dockerfile:35]: Please change this back to 3.13

File: remediation-service/Dockerfile [hunk 2]
```diff
@@ -19,44 +20,30 @@ COPY go.mod go.sum ./
...
 EXPOSE 8080
 
 # required for external tools to detect this as a go binary
 ENV GOTRACEBACK=all
 
-# KEEP THE FOLLOWING LINES COMMENTED OUT!!! (they will be included within the travis-ci build)
-#travis-uncomment ADD MANIFEST /
-#travis-uncomment COPY entrypoint.sh /
-#travis-uncomment ENTRYPOINT ["/entrypoint.sh"]
-
```
Comment (christian-kreuzberger-dtx) [inline remediation-service/Dockerfile:59]: we need those lines

Comment (warber) [reply remediation-service/Dockerfile]: done

File: remediation-service/Dockerfile (hunk 1 above)
Comment (warber) [reply remediation-service/Dockerfile]: done

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM

**Pull request merged**